SOPT-iOS/Projects/SOPT-iOS/Sources/ModuleFactory/ModuleFactory.swift

```diff
@@ -0,0 +1,32 @@
+//
+//  ModuleFactory.swift
+//  SOPT-iOS
+//
+//  Created by 양수빈 on 2022/10/06.
+//  Copyright © 2022 SOPT-iOS. All rights reserved.
+//
+
+import Foundation
+
+import Presentation
+import Network
+import Domain
+import Data
+
+protocol ModuleFactoryProtocol {
+    func makeSampleVC() -> SampleVC
+}
+
+open class ModuleFactory: ModuleFactoryProtocol {
+  static let shared = ModuleFactory()
+  private init() { }
+
+  public func makeSampleVC() -> SampleVC {
+      let repository = SampleRepository(service: BaseService.self as! SampleServiceType)
+      let useCase = DefaultSampleUseCase(repository: repository)
+    let viewModel = SampleViewModel(useCase: useCase)
+    let sampleVC = SampleVC.controllerFromStoryboard(.sample)
+    sampleVC.viewModel = viewModel
+    return sampleVC
+  }
+}
```

Comment:
생각해봤는데 이 구조를 변경하지 않고 사용하려면 Presentation에서 ModuleFactory의 interface를 가지고 있으면 될 것 같거든요? 그리고 vc의 생성자로 modulefactory를 주입해주는 식으로?

```suggestion
    let sampleVC = SampleVC(sampleFactory: self)
    sampleVC.viewModel = viewModel
```
위와 같이 할 수 있을 것 같습니다.. 이래서 코디네이터가 필요한데 없으니 이렇게라도 할 수 있을 것 같아용

Reply:
Presentation 모듈 내 에서 ModuleFactoryInterface를 생성했슴니다 @L-j-h-c @devxsby @lsj8706 
<img width="939" alt="스크린샷 2022-10-08 오후 2 29 50" src="https://user-images.githubusercontent.com/81167570/194690384-c2ec4875-4043-4a7b-b8a1-095e04bbe45a.png">

안에서 protocol로 makeSampleVC를 넣어주었구,
SOPT-iOS 모듈 내 ModuleFactory에서 Presentation 모듈의 ModuleFactoryInterface를 채택해서 VC 생성하는 코드 구현해줬슴니다
<img width="677" alt="스크린샷 2022-10-08 오후 2 29 59" src="https://user-images.githubusercontent.com/81167570/194690401-aaf69f8d-39b0-4741-94c1-00596a088673.png">

VC안에서 아래처럼 선언한 다음에 
<img width="590" alt="스크린샷 2022-10-08 오후 3 03 18" src="https://user-images.githubusercontent.com/81167570/194691794-66bee068-35d5-4d52-a1c4-86abc4697a0f.png">

아래처럼 factory.make~로 VC를 가져와서 화면 전환 할 수 잇슴니다
<img width="550" alt="스크린샷 2022-10-08 오후 3 04 13" src="https://user-images.githubusercontent.com/81167570/194691823-3f3e8dba-b3f0-4501-a44b-efb60405cdca.png">

Reply:
감사합니다 손상님.... 잘쓰겠습니다....

Reply:
양수빙수 쵝오.,,